libsupport/include/katana/Result.h

```diff
@@ -218,10 +222,58 @@ class KATANA_EXPORT [[nodiscard]] ErrorInfo {
 /// to store errors, e.g., collecting results across threads.
 class KATANA_EXPORT CopyableErrorInfo {
 public:
+  CopyableErrorInfo(const std::error_code& ec) : error_code_(ec) {}
+
+  CopyableErrorInfo() : CopyableErrorInfo(std::error_code()) {}
+
   CopyableErrorInfo(const ErrorInfo& ei);
 
+  template <typename F, typename... Args>
+  static ErrorInfo MakeWithSourceInfo(
```

Comment:
This returns `ErrorInfo` which is weird given the class it's in. This also appears to be identical to the function in `ErrorInfo`. I think this compiles because of implicit conversions.